pkg/sanitizer/helpers.go

```diff
@@ -8,38 +8,61 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
-const blocked = "blocked"
+// ClusterInfo holds the provider, capacity, and capabilities.
+type ClusterInfo struct {
+	Provider     string
+	Capacity     int
+	Capabilities []string
+}
 
-type ClusterMap map[string]string
+// ClusterMap maps a cluster name to its corresponding ClusterInfo.
+type ClusterMap map[string]ClusterInfo
 
-func LoadClusterConfig(filePath string) (ClusterMap, sets.Set[string], error) {
-	data, err := os.ReadFile(filePath)
-	if err != nil {
-		return nil, nil, err
+func loadClusterConfigFromBytes(data []byte) (ClusterMap, sets.Set[string], error) {
+	var clusters map[string][]struct {
```

Comment:
Why an anonymous struct instead of enhancing the `ClusterMap` type?
As an example:
```golang
type ClusterMap struct {
	AWS     map[string]ClusterProvider `yaml:"aws,omitempty"`
	GCP     map[string]ClusterProvider `yaml:"gcp,omitempty"`
	Blocked []string                   `yaml:"blocked,omitempty"`
}

func (cm *ClusterMap) BlockedClusters() sets.Set[string] {
	return sets.New[string](cm.BlockedClusters().UnsortedList()...)
}

type ClusterProvider struct {
	Capacity     int      `yaml:"capacity"`
	Capabilities []string `yaml:"capabilities"`
}
```

Reply:
Not sure why we should limit ourselves to gcp and aws since we discuss next cluster providers, I guess we should not create static types like this. and as for anonymous, it is used only there (it differs from what is later used and translated in the code) so no need to take it out of this function.

Reply:
> Not sure why we should limit ourselves to gcp and aws since we discuss next cluster providers

There are not so many out there that we might support, and could be added when needed.
But if you don't like "fixed" providers:

```golang
type ClusterMap struct {
	Clusters map[string]ClusterProvider `json:"providers,omitempty"`
	Blocked  []string                   `json:"blocked,omitempty"`
}

type ClusterProvider struct {
	Capacity     int      `json:"capacity"`
	Capabilities []string `json:"capabilities"`
}
```
Consider introducing something like:
```golang
func (cm *ClusterMap) BlockedClusters() sets.Set[string] {
	return sets.New[string](cm.BlockedClusters().UnsortedList()...)
}
```
because `func` returning three parameters are quite unidiomatic in Go:
```golang
LoadClusterConfig(filePath string) (ClusterMap, sets.Set[string], error)
```